### ISSUE TITLE
[java] AccessorClassGeneration: False Negative when private constructor called from lambda expression

### ISSUE THREAD

**Issue by mohui1999** [pmd/pmd#4468]
**Affects PMD Version:** 7.0.0

**Rule:** [AccessorClassGeneration](https://docs.pmd-code.org/latest/pmd_rules_java_bestpractices.html#accessorclassgeneration)

**Description:**
The `AccessorClassGeneration` rule detects when a private constructor is called from outside its class, which may cause the generation of an accessor class. 

However, PMD may not report this violation when the private constructor is called from a lambda expression, which could lead to a false negative. While it reports this violation when it is called from anonymous class.

Note: This test case was run with source-type set to Java 10.

**Code Sample demonstrating the issue:**
Maybe a false negative:

```java
public class Example_lambda extends View {
            private int foo;

            private Example(final Context context) {
                super(context);
            }

            public static void create() {
                new Handler().post(() -> {
                    Example example = new Example(null);
                });
            }
        }
```

Correctly reported code:

```java
public class Example extends View {
    private int foo;

    private Example(final Context context) {
        super(context);
    }

    public static void create() {
        new Handler().post(new Runnable() {
            @Override
            public void run() {
                Example example = new Example(null); // Should be marked since the constructor is private.
            }
        });
    }
}
```

**Expected outcome:**

PMD should report a violation at line 10, but doesn't. This is a false negative.

**Running PMD through:** *[CLI]*
